util/validateGeojson.js

```diff
@@ -1,85 +1,148 @@
 'use strict';
 
+const transform = require('parallel-transform');
 const geojsonhint = require('@mapbox/geojsonhint').hint;
-const readLineSync = require('n-readlines');
-const path = require('path');
 const turf = require('@turf/turf');
 const rewind = require('geojson-rewind');
+const Ajv = require('ajv');
+
+const ajv = new Ajv({
+    schemaId: 'auto'
+});
+
+ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
 
 /**
  * Ensure geometries are valid before import
  *
- * @param {String} filepath File to validate against
- * @param {Object} opts
+ * @param {Object} opts Options object
  * @param {boolean} opts.ignoreRHR=false Ignore Right Hand Rule errors
+ * @param {Object} opts.schema JSON Schema to validate properties against
  */
-function validateGeojson(filepath, opts = {}) {
-    // Read each feature
-    const rl = new readLineSync(filepath);
-    // Get the file name
-    const filename = path.basename(filepath);
+function validateGeojson(opts = {}) {
     // Flag to track the feature line number
     let linenumber = 0;
-    // list of errors linked to the file name and line number
-    const corruptedfeatures = [];
-
-    let line = true;
-
-    while (line) {
-        line = rl.next();
-        if (!line) break;
 
-        validateFeature(line.toString('utf8'));
+    let schema = false;
 
-        if (corruptedfeatures.length > 100) break;
+    if (opts.schema) {
+        schema = ajv.compile(opts.schema);
     }
 
-    // Validate each feature
-    function validateFeature(line) {
-        const feature = rewind(JSON.parse(line));
-        linenumber++;
-        let errors = [];
+    return transform(1, (feat, cb) => {
```

Comment:
If the parallelism is 1, do we need to use the parallel library?

Reply:
@mattciferri It makes implementing a transform stream much easier, and we can experiement with increasing this in the future